src/LiveComponent/src/LiveComponentHydrator.php

```diff
@@ -148,6 +148,8 @@ public function hydrate(object $component, array $data, string $componentName):
             if ($method = $liveProp->hydrateMethod()) {
                 // TODO: Error checking
                 $value = $component->$method($value);
+            } elseif (\is_string($value) && $type && \in_array($type->getName(), ['int', 'float', 'bool'], true)) {
+                $value = self::coerceScalarValue($value, $type);
             } elseif (!$value && $type && $type->allowsNull() && is_a($type->getName(), \BackedEnum::class, true) && !\in_array($value, array_map(fn (\BackedEnum $e) => $e->value, $type->getName()::cases()))) {
                 $value = null;
             } elseif (null !== $value && $type && !$type->isBuiltin()) {
```

Comment:
This area is increasingly feeling "ad-hoc", but it's well-covered in tests. And if there is some broader solution for handling these edge cases, it'll become obvious eventually 👍